HeterogeneousCore/SonicTriton/src/TritonService.cc

```diff
@@ -286,16 +322,35 @@ void TritonService::preBeginJob(edm::PathsAndConsumesOfModulesBase const&, edm::
         << "TritonService: Starting the fallback server failed with exit code " << rv;
   } else if (verbose_)
     edm::LogInfo("TritonService") << output;
+
+  //get the chosen device
+  std::string chosenDevice(fallbackOpts_.device);
+  if (chosenDevice == "auto") {
+    chosenDevice = extractFromLog(output, "CMS_TRITON_CHOSEN_DEVICE: ");
+    if (!chosenDevice.empty()) {
+      if (chosenDevice == "cpu")
+        server.type = TritonServerType::LocalCPU;
+      else if (chosenDevice == "gpu")
+        server.type = TritonServerType::LocalGPU;
+      else
+        throw cms::Exception("FallbackFailed")
+            << "TritonService: unsupported device choice " << chosenDevice << " for fallback server, log follows:\n"
+            << output;
+    } else
+      throw cms::Exception("FallbackFailed")
```

Comment:
I feel these exceptions would be better as
```suggestion
      throw edm::Exception(edm::errors::ExternalFailure)
```
(following https://github.com/cms-sw/cmssw/pull/43274)